pkg/execution/executor/executor.go

```diff
@@ -805,7 +808,7 @@ func (e *executor) handleGeneratorStep(ctx context.Context, gen state.GeneratorO
 	}
 
 	// Re-enqueue the exact same edge to run now.
-	if err := e.sm.Scheduled(ctx, item.Identifier, nextEdge.Incoming, 0, nil); err != nil {
+	if err := e.sm.Scheduled(ctx, item.Identifier, nextEdge.Incoming, 0, nil, item.DisableImmediateExection); err != nil {
```

Comment:
`Scheduled` seemed like the safest place to add this. Is this staying in refactors?